cmd/internal/register/maps.go

```diff
@@ -0,0 +1,170 @@
+}
+
+// Maps registers maps with with atlas
+func Maps(a *atlas.Atlas, maps []config.Map, providers map[string]provider.Tiler) error {
+
+	// iterate our maps
+	for _, m := range maps {
+		newMap := atlas.NewWebMercatorMap(string(m.Name))
+		newMap.Attribution = html.EscapeString(string(m.Attribution))
+
+		// convert from env package
+		centerArr := [3]float64{}
+		for i, v := range m.Center {
+			centerArr[i] = float64(v)
+		}
+
+		newMap.Center = centerArr
+
+		if len(m.Bounds) == 4 {
+			newMap.Bounds = geom.NewExtent(
+				[2]float64{float64(m.Bounds[0]), float64(m.Bounds[1])},
+				[2]float64{float64(m.Bounds[2]), float64(m.Bounds[3])},
+			)
+		}
+
+		// iterate our layers
+		for _, l := range m.Layers {
+			// split our provider name (provider.layer) into [provider,layer]
+			providerLayer := strings.Split(string(l.ProviderLayer), ".")
+
+			// we're expecting two params in the provider layer definition
+			if len(providerLayer) != 2 {
+				return ErrProviderLayerInvalid{
+					ProviderLayer: string(l.ProviderLayer),
+					Map:           string(m.Name),
+				}
+			}
+
+			// lookup our proivder
+			provider, ok := providers[providerLayer[0]]
+			if !ok {
+				return ErrProviderNotFound{providerLayer[0]}
+			}
+
+			// read the provider's layer names
+			layerInfos, err := provider.Layers()
+			if err != nil {
+				return ErrFetchingLayerInfo{
+					Provider: providerLayer[0],
+				}
+			}
+
+			// confirm our providerLayer name is registered
+			var found bool
+			var layerGeomType tegola.Geometry
+			for i := range layerInfos {
+				if layerInfos[i].Name() == providerLayer[1] {
+					found = true
+
+					// read the layerGeomType
+					layerGeomType = layerInfos[i].GeomType()
+				}
+			}
+			if !found {
+				return ErrProviderLayerNotRegistered{
+					MapName:       string(m.Name),
+					ProviderLayer: string(l.ProviderLayer),
+					Provider:      providerLayer[0],
+				}
+			}
+
+			var defaultTags map[string]interface{}
+			if l.DefaultTags != nil {
+				var ok bool
+				defaultTags, ok = l.DefaultTags.(map[string]interface{})
+				if !ok {
+					return ErrDefaultTagsInvalid{
+						ProviderLayer: string(l.ProviderLayer),
+					}
+				}
+			}
+
+			var minZoom uint
+			if l.MinZoom != nil {
+				minZoom = uint(*l.MinZoom)
+			}
+
+			var maxZoom uint
+			if l.MaxZoom != nil {
+				maxZoom = uint(*l.MaxZoom)
+			}
+
+			// add our layer to our layers slice
+			newMap.Layers = append(newMap.Layers, atlas.Layer{
+				Name:              string(l.Name),
+				ProviderLayerName: providerLayer[1],
+				MinZoom:           minZoom,
+				MaxZoom:           maxZoom,
+				Provider:          provider,
+				DefaultTags:       defaultTags,
+				GeomType:          layerGeomType,
+				DontSimplify:      bool(l.DontSimplify),
+			})
+		}
+
+		// register the map. if an atlas is provided then use it, else use the default atlas
+		if a != nil {
+			a.AddMap(newMap)
+		} else {
+			atlas.AddMap(newMap)
+		}
```

Comment:
a `nil` atlas will already add it to the [defaultAtlas](https://github.com/go-spatial/tegola/blob/6b0e9ce614c363073805fa8096079754a21afba0/atlas/atlas.go#L153), so this check is not needed.